### ISSUE TITLE
Bug in computing the quaternion representation of a rotation.

### ISSUE THREAD

**Issue by jacarvalho** [facebookresearch/theseus#343]
## 🐛 Bug

The method `to_quaternion` in `theseus.SO3` outputs a wrong quaternion.

## Steps to Reproduce

```python
import torch
import theseus as th
import numpy as np


def to_torch(x, dtype=torch.float, device='cpu', requires_grad=False):
    return torch.tensor(x, dtype=dtype, device=device, requires_grad=requires_grad)


# Rotation matrix expected
rot_mat_des = to_torch(
    np.array([[0, 1, 0.],
              [1, 0, 0],
              [0, 0, -1]])
).view(1, 3, 3)

# Rotation quaternion expected
rot_quat_des = to_torch(np.array([0, 0.7071068, 0.7071068, 0])).view(1, 4)

######
# Rotation object instantiated with a rotation matrix
rot = th.SO3(tensor=rot_mat_des)

# Rotation matrix computed with theseus
rot_mat = rot.to_matrix()
print(f"Rotation matrix theseus: {rot_mat}")
print(f"Rotation matrix desired: {rot_mat_des}")
print(f"All close: {torch.allclose(rot_mat, rot_mat_des, rtol=1e-05, atol=1e-05)}")
print()

# Rotation quaternion computed with theseus
rot_quat = rot.to_quaternion()
print(f"Rotation quaternion theseus: {rot_quat}")
print(f"Rotation quaternion desired: {rot_quat_des}")
print(f"All close: {torch.allclose(rot_quat, rot_quat_des, rtol=1e-05, atol=1e-05)}")
print()

######
# Rotation object instantiated with a rotation quaternion
rot = th.SO3(quaternion=rot_quat_des)

# Rotation matrix computed with theseus
rot_mat = rot.to_matrix()
print(f"Rotation matrix theseus: {rot_mat}")
print(f"Rotation matrix desired: {rot_mat_des}")
print(f"All close: {torch.allclose(rot_mat, rot_mat_des, rtol=1e-05, atol=1e-05)}")
print()

# Rotation quaternion computed with theseus
rot_quat = rot.to_quaternion()
print(f"Rotation quaternion theseus: {rot_quat}")
print(f"Rotation quaternion desired: {rot_quat_des}")
print(f"All close: {torch.allclose(rot_quat, rot_quat_des, rtol=1e-05, atol=1e-05)}")
print()
```

## Expected behavior

I'd expect that when creating a rotation with `theseus.SO3` with a valid rotation matrix, I'd get the correct quaternion representation of this rotation.

## Additional context

In practice maybe one does not have a rotation matrix where the axes are perfectly aligned with the origin reference frame (entries with zero values), but this is an unexpected behavior.

## System Info

- OS (e.g., Linux): Ubuntu 20.04
- Python version: 3.8.13
- GPU models and configuration: NVIDIA GeForce MX150
- CUDA version: 11.8
- pip/conda dependencies packages versions: pytorch, numpy

**Comment by fantaosha** [comment]
Hello, thanks for reporting this. The bug is fixed at #344

**Comment by jacarvalho** [comment]
Perfect. Thank you.
